### PR TITLE
xdg-autostart-service: Add comments to tilde expansion

### PR DESCRIPTION
**Pull request by davidedmundson**
There were follow up review comments in my previous PR about ensuring xdg-autostart retro-compatibility

### CODE REVIEW

**Review by bluca — APPROVED** [review]

File: src/xdg-autostart-generator/xdg-autostart-service.c [hunk 1]
```diff
@@ -443,7 +443,11 @@ int xdg_autostart_format_exec_start(
                         return log_oom();
 
                 /*
-                 * Expand ~ if it comes at the beginning of an argument to form a path
+                 * Expand ~ if it comes at the beginning of an argument to form a path.
+                 *
+                 * This is not explicitly in the desktop specification,
```
Comment (yuwata) [inline src/xdg-autostart-generator/xdg-autostart-service.c]: Maybe, "explicitly supported" or "explicitly documented"?

File: src/xdg-autostart-generator/xdg-autostart-service.c [hunk 2]
```diff
@@ -443,7 +443,11 @@ int xdg_autostart_format_exec_start(
...
+                 * but needed for compatibility with former KDE code
+                 * which supported a more shell-like syntax for users making custom entries
```
Comment (keszybz) [inline src/xdg-autostart-generator/xdg-autostart-service.c]: Also dot at the end of the sentence here.

File: src/xdg-autostart-generator/xdg-autostart-service.c [hunk 3]
```diff
@@ -443,7 +443,11 @@ int xdg_autostart_format_exec_start(
...
                  */
```
Comment (poettering) [inline src/xdg-autostart-generator/xdg-autostart-service.c:450]: btw, doesn't really matter, but we suggest line breaks at 109 chars... see coding style, .editorconfig, .vimrc, .elisp, .dir-localy....

File: src/xdg-autostart-generator/xdg-autostart-service.c (hunk 1 above)
Comment (poettering) [reply src/xdg-autostart-generator/xdg-autostart-service.c]: I'd drop "explicitly"... I mean, it's not "implicitly" in the spec either afaics.

Just say "the specification does not mandate this, but we do it anyway for compatibility"

**Comment by bluca** [comment]
The reviews were trivial changes to the comment, so I've just pushed an update, so that we can get this in 252-rc2.

**Pull request merged**